deeposlandia/__main__.py

```diff
@@ -0,0 +1,293 @@
+        default=0,
+        help=("Number of testing images"),
+    )
+    parser.set_defaults(func=reference_func)
+
+
+def train_parser(subparser, reference_func):
+    """Add instance-specific arguments from the command line
+
+    Parameters
+    ----------
+    subparser : argparser.parser.SubParsersAction
+    reference_func : function
+    """
+    parser = subparser.add_parser(
+        "train",
+        help="Train convolutional neural networks for automated image analysis"
+    )
+    add_dataset_args(parser)
+    add_nb_image_args(parser)
+    add_training_args(parser)
+    parser.add_argument(
+        "-s",
+        "--image-size",
+        default=256,
+        type=int,
+        help=("Desired size of images (width = height)"),
+    )
+    parser.add_argument(
+        "-e",
+        "--nb-epochs",
+        type=int,
+        default=0,
+        help=(
+            "Number of training epochs (one epoch means "
+            "scanning each training image once)"
+        ),
+    )
+    parser.set_defaults(func=reference_func)
+
+
+def inference_parser(subparser, reference_func):
+    """Add instance-specific arguments from the command line
+
+    Parameters
+    ----------
+    subparser : argparser.parser.SubParsersAction
+    reference_func : function
+    """
+    parser = subparser.add_parser(
+        "infer",
+        help="Predict image labelling with a neural network model"
+    )
+    add_dataset_args(parser)
+    add_training_args(parser)
+    parser.add_argument(
+        "-i",
+        "--image-paths",
+        required=True,
+        nargs="+",
+        help="Path of the image on the file system",
+    )
+    parser.set_defaults(func=reference_func)
+
+
+def postprocess_parser(subparser, reference_func):
+    """Add instance-specific arguments from the command line
+
+    Build a high-resolution image labelling by predicting semantic segmentation
+    labels on image patches, and by postprocessing resulting arrays so as to
+    get geographic entities.
+
+    Parameters
+    ----------
+    subparser : argparser.parser.SubParsersAction
+    reference_func : function
+    """
+    parser = subparser.add_parser(
+        "postprocess",
+        help=(
+            "Postprocess the neural network prediction "
+            + "(for geographic datasets only)"
+        )
+    )
+    add_dataset_args(parser, available_datasets=GEOGRAPHIC_DATASETS)
+    parser.add_argument(
+        "-b",
+        "--batch-size",
+        default=2,
+        type=int,
+        help=("Number of images in each inference batch"),
+    )
+    parser.add_argument(
+        "-i",
+        "--image-basename",
+        required=True,
+        help="Basename of the image within the dataset",
+    )
+    parser.add_argument(
+        "-s",
+        "--image-size",
+        required=True,
+        type=int,
+        help="Image patch size, in pixels",
+    )
+    return parser
+
+
+def add_dataset_args(parser, available_datasets=AVAILABLE_DATASETS):
```

Comment:
The default value `available_datasets` is a list. You shouldn't have this type for default function values. I suggest a tuple which is a immutable python type.

Reply:
I changed the nature of `AVAILABLE_DATASETS` (and `GEOGRAPHIC DATASETS` and `AVAILABLE_MODELS` which have the same nature) to comply with your remark. Thanks!